### ISSUE TITLE
Increase information density on small (laptop) screens

### ISSUE THREAD

**Issue by nicoburns** [rust-lang/beta.rust-lang.org#412]
# Increase information density and add narative
## Summary
I would like to modify the overall design philosophy to more information dense, and to make use of narrative text.

## Motivation
I'm browsing the new Rust site on a 13 inch MacBook Pro (1400x900), and it's very difficult to actually glean any information from it, because there is so little information on each screen. For example compare the "Learn" pages from beta.rust-lang.org and python.org.

The rust design looks clean and modern, but there is much less actual content on screen. Additionally, the python page can be read beginning to end, with the text indicating guiding readers as to which resources might be most appropriate for them, whereas the rust page just says what the linked documentation is. **I** have more context, and know which of the links I want, but I think we could do more for users who are new to the Rust ecosystem.

https://beta.rust-lang.org/learn

<img width="1439" alt="screenshot 2018-11-29 at 18 39 16" src="https://user-images.githubusercontent.com/1007307/49244155-045dba80-f407-11e8-89c7-61cfaa08802f.png">

https://www.python.org/about/gettingstarted

<img width="1440" alt="screenshot 2018-11-29 at 18 39 27" src="https://user-images.githubusercontent.com/1007307/49244177-17708a80-f407-11e8-8d39-e9982a9593b1.png">

## Drawbacks
It could make the design looked cluttered. It's my opinion that there is still plenty of room to increase density before this becomes a problem, but it should definitely be considered.

## Rationale and alternatives
- This would make it easier for users to find what they are looking for on the website.

## Maintenance
This shouldn't require any ongoing maintenance once the change is made, although there may be work for translators if more text is used.

If people agree with this proposal, then I might be able to find some time to work on a design that implements these suggestions...

**Comment by nicoburns** [comment]
An example of how the current home page could look simply by removing whitespace:

<img width="1440" alt="screenshot 2018-11-29 at 21 28 48" src="https://user-images.githubusercontent.com/1007307/49255101-e8681200-f422-11e8-9ba7-ac65b8d216ee.png">

**Comment by ashleygwilliams** [comment]
this is a good related issue to bring up the white space but in the interest of small actionable issues, i would prefer if we could file issues per change. so i would welcome an issue to reduce the whitespace under the slogan atm. that way i (or someone else) can file a PR and close the issue.

this issue is good but much much larger and not likely to be something we can address this year.

**Comment by ashleygwilliams** [comment]
@chriskrycho as part of the tachyons migration, could u keep an eye on how we can improve the density in some places? seems like they could go together well as we work on the vertical rhythm of the pages. thanks!

**Comment by nicoburns** [comment]
It seems like it would probably be a good idea to establish standards for how much padding is left between elements. I find this is often one of the main things which makes things like "well designed".

**Comment by ashleygwilliams** [comment]
related: https://github.com/rust-lang/beta.rust-lang.org/issues/446

**Comment by chriskrycho** [comment]
#535 begins to address this. See also #525, which is next on my list of things to tackle.

**Comment by eddyp** [comment]
I strongly agree with the idea, at work I have a laptop and 2 external small (19 inch) monitors and this is how the beta page looks when Firefox is maximized on:

![image](https://user-images.githubusercontent.com/123772/49595285-4eebb380-f980-11e8-954e-cc1e4848389a.png)

This is it, basically none of the "Why Rust?" content, not even the titles are visible on the default landing page.

To have the "Why Rust?" section on the page I have to zoom out at 70%:

![image](https://user-images.githubusercontent.com/123772/49595548-e0f3bc00-f980-11e8-9738-a26c691f4cc2.png)

Even at 80% the last rows of the "Reliability" and "Performance" columns are out of the screen.

From the PoV of aestetics, I think some possible solutions are:
 - decrease the font sizes for the "Rust" text, the motto and all subtitles.
 - decrease a little the gap before "Rust" in the top part (with the logo and the Install Learn ... links)
   - by decreasing the Rust logo, it seems the top part is taking a biger part of the page than the links "pushing" the "Rust" text unnecessarily lower on the page.
- Move the "Why Rust?" text up by decreasing the gaps before and after text items in the zones after the "Rust" zone (e.g. the green-ish background zone, the purple-ish background zone etc.)

**Comment by chriskrycho** [comment]
I'm not sure what resolution you're dealing with, but even at 1280⨉720, most of that first section is visible:

<img width="640" alt="screen shot 2018-12-06 at 10 01 40 am" src="https://user-images.githubusercontent.com/2403023/49599364-0ad7bf00-f93e-11e8-9249-15dca8c188ec.png">

We could certainly tweak it slightly further, but I think that's a pretty reasonable place to be at the moment. 🤔

**Comment by eddyp** [comment]
I am using 2 Dell E1911 whose maximum resolution is 1440x900.

**Comment by eddyp** [comment]
@chriskrycho I think I know *why* on my screen on Windows there so little content compared to yours. The problem is that Windows 10 has been a pain to configure to make true type fonts look comparable to what Windows 7 was doing properly and the only solution to not make my eyes hurt is to increase the font size to 125% on all monitors. This is not just one case, all of my work colleagues found this as the only solution that gives true type fonts in Windows 10 which do not break on reset or when disconnecting the laptops from docking stations and connecting external projectors.

**Comment by eddyp** [comment]
On an identical laptop in the same setup, but without the true type font workaround the current page is a little bit more decent, thanks @chriskrycho for the hint. 

![image](https://user-images.githubusercontent.com/123772/49729455-3203f880-fc7e-11e8-8a23-263c07a00995.png)

Still, I'd consider decreasing the gaps/fonts so the "Why Rust?" part doesn't get cut off.

**Comment by mitsuhiko** [comment]
Not sure if this belongs into this issue but the current font size feels quite a bit too large.

**Comment by chriskrycho** [comment]
@mitsuhiko I appreciate the feedback, but I'm afraid we're probably not going to tweak the overall font size again. We've landed on a midpoint for the font size which feels *too large* for some people and *barely big enough* for others, and is hitting our targets for legibility. We may make specific tweaks to specific points in the site, but not to the overall size.

**Comment by mitsuhiko** [comment]
@chriskyrcho I honestly doubt that you found the sweet point. I’m sure if you were to run a survey on the font size the vast majority of respondents would reply with it being too large.

I can’t even find comments asking for larger fonts. The problem was that too much information was visible at once.
